File: lesunny/048_크레인인형뽑기게임.java
import java.util.ArrayList;

class Solution {
    public int checkNpop(ArrayList<Integer> basket){        
        for(int i=0; i<basket.size(); i++){
            if(i!=basket.size()-1 && basket.get(i)==basket.get(i+1)){
                basket.remove(i);
                basket.remove(i);
                return 2;
            }else if(i==basket.size()-1){
                return 0;
            }
        }
        return 0;
    }
    
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        ArrayList<Integer> basket = new ArrayList<>();
        
        for(int i=0; i<moves.length; i++){
            for(int j=0; j<board.length; j++){
                if(board[j][moves[i]-1] !=0){
                    basket.add(board[j][moves[i]-1]);
                    board[j][moves[i]-1] = 0;
                    answer += checkNpop(basket);
                    break;
                }
            }
            
        }
        
        return answer;
    }
}

File: JIWON_CODE/048 크레인 인형뽑기 게임.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    int answer = 0;
    vector<int> result;
    
    for(int i=0;i<moves.size();i++){
        for(int j=0;j<board.size();j++){
            if(board[j][moves[i]-1] != 0){
                result.push_back(board[j][moves[i]-1]);
                
                board[j][moves[i]-1] = 0;
                break;
            }
        }
    }
    
    for(int k = 1;k<result.size();k++){
        if(result[k] == result[k-1]){
            answer = answer + 2;
            result.erase(result.begin() + --k);
            result.erase(result.begin() + k++);
            
            k = 0;
        }
    }
    
    return answer;
}
